agent/doc/swdesign/README.md

```diff
@@ -288,6 +297,101 @@ The following diagram show the general steps the Ankaios Agent takes when receiv
 
 ![Handling UpdateWorkload](plantuml/seq_update_workload.svg)
 
+#### Agent handles UpdateWorkload requests from the server
+`swdd~agent-handles-update-workload-requests~1`
+
+Status: approved
+
+When the AgentManager receives an `UpdateWorkload` message from the server, then the RuntimeManager shall handle the workloads contained within the message.
+
+Comment: The `UpdateWorkload` message contains workloads to create, update and delete.
+
+Rationale: This ensures separation between receiving the requests from the server and performing the requested actions.
+
+Tags:
+- AgentManager
+- RuntimeManager
+
+Needs:
+- impl
+- utest
+
+#### RuntimeManager transforms UpdateWorkload message into WorkloadOperations
+`swdd~agent-transforms-update-workload-message-to-workload-operations~1`
+
+Status: approved
+
+When the RuntimeManager receives the workloads of an `UpdateWorkload` message, then the RuntimeManager shall transform the workloads inside the message into a list of `WorkloadOperation`s containing workloads to create, update and delete.
+
+Comment: The list of `WorkloadOperation`s contains the actions on the workloads which the RuntimeManager shall perform.
+
+Rationale: The inter-workload dependency handling requires the concrete information about the type of operation performed on the workload.
+
+Tags:
+- RuntimeManager
+- WorkloadOperation
+
+Needs:
+- impl
+- utest
+
+#### RuntimeManager executes create workload operation
+`swdd~agent-executes-create-workload-operation~1`
+
+Status: approved
+
+When the RuntimeManager receives the `WorkloadOperation` to create a workload, then the RuntimeManager shall request the RuntimeFacade to create the workload.
```

Comment:
Shouldn't we mention the precondition that the dependencies must be fulfilled?

Reply:
Isn't it then the old problem again that the RuntimeManager does not know about when the dependencies are fulfilled?

What about mentioning it inside a comment of this reqs (and in the similar ones) or we connect here again the WorkloadScheduler by saying explicitly that the workload operations are received from it. Then it is according to the reqs `swdd~agent-handles-workloads-with-fulfilled-dependencies~1` and `swdd~agent-handles-new-workload-operations` saying already that the RuntimeManager executes the received list and the reqs here are saying how this is done with details. The only thing I see then is that we need to tag the WorkoadScheduler again inside this req.

Reply:
I completely agree with you. We have mention already that the RuntimeManager receives the list of operations to be executed according to `swdd~agent-handles-workloads-with-fulfilled-dependencies~1`, I see the problem here that, as you mention, we also have another list of operations `swdd~agent-handles-new-workload-operations` and it is not explicitly specified which list is handled. As we have already defined the list of workloads with fulfilled inter-workload dependencies in `swdd~agent-handles-workloads-with-fulfilled-dependencies~1`, we don't need to do it again. In my eyes it is already sufficient here to mention the "fulfilled inter-workload dependencies":

```suggestion
When the RuntimeManager receives a create `WorkloadOperation` with fulfilled inter-workload dependencies, then the RuntimeManager shall request the RuntimeFacade to create the workload.
```

Reply:
The same must be done in the corresponding requirements below.

Reply:
Ok thank you, I will adapt the three reqs and commit.

Reply:
I have adapted the reqs. I was just thinking about if I need to tag the WorkloadObject really with this. The WorkloadObject is mentioned inside the req texts, but it's more the call of the WorkloadObject not it's internal functionality of the triggered method. Therefore, I have decided to tag it only on the caller side in the runtime manager `workload_object.method...`.